File: src/main/java/br/com/hanrry/inventory/entity/Product.java
package br.com.hanrry.inventory.entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@EqualsAndHashCode
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "product")
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, unique = true)
    private String sku;

    @Column(name = "min_stock", nullable = false)
    private Long minStock;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL)
    private List<Batch> batchs = new ArrayList<>();
}

File: src/services/contracts/product.ts
export interface Product {
  id: number
  name: string
  sku: string
  totalQuantity: number
  categoryName: string
  minStock: number
}

export interface ProductInput {
  name: string
  sku: string
  minStock: number
  categoryId: number
}

export type ProductUpdateInput = Pick<ProductInput, 'name' | 'minStock'>
